lib/apiservers/engine/backends/container.go

```diff
@@ -287,6 +287,11 @@ func (c *ContainerBackend) ContainerExecInspect(eid string) (*backend.ExecInspec
 	}
 
 	exit := int(ec.ExitCode)
+	if ec.State == constants.TaskFailedState {
+		// this is a docker specific case, specifically when the binary does not exist and we are not detached docker client returns 126
```

Comment:
This is the docker personality so almost everything is docker specific. There is also a convert/state.go or similar that’s used to do this kind of injection for start/attach. Don’t know if it should be unified with this.
Also of note is that this will impact the detached execs as well if queries via the API, and you’re not filtering on attach status so the comment is further confusing.
You can simply say `docker expects 126 for no such executable`.

You should also test trying to run a file that does exist but the user doesn’t have permission to execute, and then one that’s not an executable at all. I suspect you’ll find the exits actually mean things.